### DISCUSSION TITLE
In line or 'local' variable option when making matches?

### DISCUSSION THREAD

**Discussion by elminster3k** [scandum/tintin#227] [Script help]
When doing pattern captures, I'm wondering if there is a way to set the variable during the capture, maybe thought of as a local scope variable, for example, rather than this:
{My name is %1} {#var name %1}
It would be something like this:
{My name is {$name}}   --  Just making up syntax here, since I'm not sure what this would look like..
So in the second example, the variable is defined as it makes the match, scoped locally for that action only etc.

**Comment by scandum** [comment]
It's not possible, if you want to assign variables in a substitution you could use something like this:
```
#sub {My name is %1} {My name is @setvar{name;%1}}

#function setvar {#var {%1} {%2};#return {%2}}
```
